src/server/scripts/Northrend/Naxxramas/boss_maexxna.cpp

```diff
@@ -151,7 +185,48 @@ class boss_maexxna : public CreatureScript
         void JustDied(Unit*  killer) override
         {
             BossAI::JustDied(killer);
-            summons.DespawnAll();
+        }
+
+        void DoCastWebWrap()
+        {
+            std::list<Unit*> candidates;
+            SelectTargetList(candidates, RAID_MODE(1, 2), SelectTargetMethod::Random, 1, WebTargetSelector(me));
+
+            std::vector<uint32> positions {0, 1, 2, 3, 4, 5, 6};
+            Acore::Containers::RandomShuffle(positions);
+
+            if (candidates.empty())
+                return;
+
+            for (int i = 0; i < RAID_MODE(1, 2) ; i++)
+            {
+                if (candidates.empty())
+                    break;
+                const Position &randomPos = PosWrap[positions[i]];
+
+                auto itr = candidates.begin();
+
+                if (candidates.size() > 1)
+                    std::advance(itr, urand(0, candidates.size() - 1));
+
+                Unit *target = *itr;
+                candidates.erase(itr);
+
+                float dx = randomPos.GetPositionX() - target->GetPositionX();
+                float dy = randomPos.GetPositionY() - target->GetPositionY();
+                float distXY = sqrt((dx * dx) + (dy * dy));
```

Comment:
there are a standard function - hypot
https://en.cppreference.com/w/cpp/numeric/math/hypot

Reply:
Ty
```
                float distXY = std::hypotf(dx, dy);
```
This should do?

Reply:
yes  ^_^